src/main/java/org/spongepowered/api/world/storage/WorldProperties.java

```diff
@@ -219,6 +220,33 @@
      */
     PortalAgentType getPortalAgentType();
 
+    /**
+     * Gets the portal agent map of this world.
+     * Keys represent dimensions names.
+     * Values represent destination world names.
+     *
+     * @return The portal agent map of this world
+     */
+    Map<String, String> getPortalAgents();
+
+    /**
+     * Set the destination world for a {@link PortalType}
+     * on this world.
+     *
+     * @param portalType The type of the portal
+     * @return The name of the destination world
+     */
+    String getPortalDestination(PortalType portalType);
+
+    /**
+     * Set the destination world for a {@link PortalType}
+     * on this world.
+     *
+     * @param portalType The type of the portal
+     * @param destinationWorldName The destination world name of the portal
+     */
+    void setPortalDestination(PortalType portalType, String destinationWorldName);
```

Comment:
Again, with my above comments. I think this should be a `WorldProperties` destination world, which is fixed up by the implementation.

Reply:
This one can still be a String else you would query the server for a world just to set a string and then be done.

Reply:
Also, it SHOULD allows String values for unknown worlds currently.

Reply:
It's not clear here. Do we have a compromise on String or WorldProperties as parameter for destination ?